packages/reporters/dev-server/src/Server.js

```diff
@@ -149,29 +149,29 @@ export default class Server {
   sendIndex(req: Request, res: Response) {
     if (this.bundleGraph) {
       // If the main asset is an HTML file, serve it
-      let htmlBundle = this.bundleGraph.traverseBundles(
-        (bundle, context, {stop}) => {
-          if (bundle.type !== 'html' || !bundle.isEntry) return;
-
-          if (!context) {
-            context = bundle;
-          }
-
-          if (
-            context &&
-            bundle.filePath &&
-            bundle.filePath.endsWith('index.html')
-          ) {
-            stop();
-            return bundle;
-          }
-        },
-      );
+      let htmlBundleFilePaths = [];
+      this.bundleGraph.traverseBundles(bundle => {
+        if (bundle.type !== 'html' || !bundle.isEntry) {
+          return;
+        }
+
+        htmlBundleFilePaths.push(bundle.filePath);
+      });
 
-      if (htmlBundle) {
+      let indexFilePath =
+        htmlBundleFilePaths.length > 1
+          ? htmlBundleFilePaths
+              .sort((a, b) => {
+                return a.length - b.length;
+              })
```

Comment:
I'm guessing you did this so that index files that aren't in subdirectories are preferred?

Maybe the strings themselves should be sorted as well, the order might not be stable across builds.

Reply:
Yeah it's to ensure the most upper level index file gets served, added a second sort for sorting alphabetically as well (if length is the same